### PR TITLE
Chore: API stores subscription information on users

### PR DESCRIPTION
**Pull request by maxarvid**
Link to PT: https://www.pivotaltracker.com/story/show/181884772

```
As an API
In order to prepare for monetization for the platform
I would like to provide an endpoint to register if the user is subscribed
```

### CODE REVIEW

File: spec/requests/api/subscriptions/update_spec.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+RSpec.describe 'POST /api/subscriptions', type: :request do
+  let(:user) { create(:user) }
+  let(:credentials) { user.create_new_auth_token }
+
+  subject { response }
+
+  describe 'as a authenticated user, when subscribing' do
+    before do
+      post '/api/subscriptions', params: { user: { id: user.id } }, headers: credentials
```
Comment (tochman) [inline spec/requests/api/subscriptions/update_spec.rb]: Do you really need to pass i n the user id in params? The request contains the user info in headers.

File: spec/requests/api/subscriptions/update_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    end
+
+    it { is_expected.to have_http_status 201 }
+
+    it 'is expected to return a message that subscription status was changed' do
+      expect(response_json['message']).to eq 'You are now subscribed.'
+    end
+
+    it 'is expected that the user subscription status is changed' do
+      expect(User.last.subscribed).to eq true
```
Comment (tochman) [inline spec/requests/api/subscriptions/update_spec.rb]: Consider: 

```ruby
user.reload
expect(user.subscribed).to eq true
```

File: spec/requests/api/subscriptions/update_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
+RSpec.describe 'POST /api/subscriptions', type: :request do
+  let(:user) { create(:user) }
+  let(:credentials) { user.create_new_auth_token }
+
+  subject { response }
+
+  describe 'as a authenticated user, when subscribing' do
```
Comment (tochman) [inline spec/requests/api/subscriptions/update_spec.rb]: Consider removing `, when subscribing` from this message. My point is that it is redundant.

File: spec/requests/api/subscriptions/update_spec.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    it 'is expected to return a message that subscription status was changed' do
+      expect(response_json['message']).to eq 'You are now subscribed.'
+    end
+
+    it 'is expected that the user subscription status is changed' do
+      expect(User.last.subscribed).to eq true
+    end
+  end
+
+  describe 'as an unauthenticated user, when subsribing' do
```
Comment (tochman) [inline spec/requests/api/subscriptions/update_spec.rb]: See comment above

File: spec/requests/api/subscriptions/update_spec.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    before do
+      post '/api/subscriptions', params: { user: { id: user.id } }, headers: nil
```
Comment (tochman) [inline spec/requests/api/subscriptions/update_spec.rb]: See commet above about passing in use info in params

File: spec/requests/api/subscriptions/update_spec.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    end
+
+    it { is_expected.to have_http_status 401 }
+
+    it 'is expected to return an error message' do
+      expect(response_json['errors'][0]).to eq 'You need to sign in or sign up before continuing.'
+    end
+
+    it 'is expected not to change the user subscription status' do
+      expect(User.last.subscribed).to eq false
```
Comment (tochman) [inline spec/requests/api/subscriptions/update_spec.rb]: Consider `user.reload` here as well (see comment above)

File: app/controllers/api/subscriptions_controller.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+class Api::SubscriptionsController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    user = User.find(params[:user][:id])
+    user.subscribed = true
+    user.save
```
Comment (tochman) [inline app/controllers/api/subscriptions_controller.rb]: You should make use of Devise and go with: 

```ruby
current_user.update(subscribed: true) 
```

**Review by tochman — CHANGES_REQUESTED** [review]
Please see comments

File: spec/requests/api/subscriptions/update_spec.rb (hunk 1 above)
Comment (maxarvid) [reply spec/requests/api/subscriptions/update_spec.rb]: Indeed we do not, it's been removed

File: spec/requests/api/subscriptions/update_spec.rb (hunk 5 above)
Comment (maxarvid) [reply spec/requests/api/subscriptions/update_spec.rb]: Removed

File: spec/requests/api/subscriptions/update_spec.rb (hunk 2 above)
Comment (maxarvid) [reply spec/requests/api/subscriptions/update_spec.rb]: Refactored as a chained one-liner

File: spec/requests/api/subscriptions/update_spec.rb (hunk 6 above)
Comment (maxarvid) [reply spec/requests/api/subscriptions/update_spec.rb]: Refactored as above

**Review by tochman — APPROVED** [review]
Great job!

**Pull request merged**